### DISCUSSION TITLE
Having Issue, because of caching routes

### DISCUSSION THREAD

**Discussion by SaiRev0** [vercel/next.js#61286] [Help]
### Summary

I have a application in which I have google login with next-auth and prisma, I m using jwt token in the middleware to check for the token details. I want that once the user logins with the goofle account, I redirect the new users to a register page, where I ask for further details. The issue is that if the user tries to go back to the home page without registering, the user is signed up but I still need the details so if the user tries to go to a protected route the middleware redirectes the user to the register page. The issue arrises here is that after registering if the user goes to the protected route again the route is cached to the register page and the router skips the middleware and goes to the register page again. I m tired with this error I have tried every solution presented, like

setting x-middleware-cache as no cache
adding the http status code to 303 or 307
adding prefetch={false} in the link

Nothing works. Please help me fix the issue, please

I m attching the repo here, 
[repo](https://github.com/SaiRev0/AasPaas-Nextjs)

In that the protected routes are in 

- `/src/app/product/new/page.tsx` and few others

- `/src/middleware.tsx` the middleware

- `/src/components/header/user/UserAccountBox.tsx` has the links to these routes

- `/src/lib/auth.ts` contains the options from the next-auth

### Additional information

```js
Deployed project link:- https://www.aaspaas.tech
```

### Example

https://www.aaspaas.tech/

**Comment by makibat** [comment]
Hey!

There are a couple of ways you can clear your cache, depending on your exact needs. There was a [post](https://github.com/vercel/next.js/discussions/54075) that went into great detail about how each of the purging functions behave with some examples of their common use cases, so definitely take a look at that.

I recommend you specifically look into `revalidatePath` because that's what's usually recommended for auth flows:

> Login / logout
> - You submit the username/password (or other auth flow) through a form that is backed by a server action. You set a cookie, and then call revalidatePath / revalidateTag to ensure the Router Cache is purged and a new render happens with the newly set cookie.

Also, calling `router.refresh()` from a client component invalidates all caches including **back/forward**, so try that as well if for some reason you can't make use of the former.

> The main thing to keep in mind with this behavior is that the client-side Component Cache that is used when navigating back/forward is only purged when you call one of the purging functions (router.refresh(), revalidatePath, revalidateTag).

Hope that helps!

**Reply by SaiRev0** [reply]
I tried adding the `router.refresh()`, but it did not help. At last, I just attached a query string that changes every time as the workaround for the caching behaviour.
